src/Common/Thermodynamics/config_numerical_method.jl

```diff
@@ -0,0 +1,119 @@
+# These functions (variants of sa_numerical_method)
+# return an instance of a numerical method to solve
+# saturation adjustment, for different combinations
+# of thermodynamic variable inputs.
+
+# @print only accepts literal strings, so we must
+# branch to print which method is being used.
+function print_numerical_method(
+    ::Type{sat_adjust_method},
+) where {sat_adjust_method}
+    if sat_adjust_method <: NewtonsMethod
+        @print("    Method=NewtonsMethod")
+    elseif sat_adjust_method <: NewtonsMethodAD
+        @print("    Method=NewtonsMethodAD")
+    elseif sat_adjust_method <: SecantMethod
+        @print("    Method=SecantMethod")
+    elseif sat_adjust_method <: RegulaFalsiMethod
+        @print("    Method=RegulaFalsiMethod")
+    else
+        error("Unsupported numerical method")
+    end
+end
```

Comment:
@vchuravy is there a cleaner way?

Reply:
Yeah we should implement `@printf` proper https://github.com/JuliaGPU/KernelAbstractions.jl/issues/73
cc: @leios